### DISCUSSION TITLE
How to use AWS Amplify created Dynamodb database with mindsdb

### DISCUSSION THREAD

**Discussion by ghost** [mindsdb/mindsdb#5535] [Get help]
I'm trying to use Mindsdb with AWS Amplify in an app. I'll be creating a Dynamodb database using `amplify add api`, and I'll be configuring authentication using the @auth directive of the Amplify CLI. The command creates a Dynamodb database, and I'll like to use that database with the Mindsdb cloud. I don't know what to do, really. Do I have to make the tables public for Mindsdb to have access to them? Like [this](https://docs.amplify.aws/cli/graphql/authorization-rules/#public-data-access). Maybe it's not needed.

I also noticed that, to link a Dynamodb table to the Mindsdb cloud, I must generate an access key and a secret AWS key. Should I create another IAM user for Mindsdb and generate those credentials for the user? And that'll work without me changing any setting in my Amplify Graphql schema? Please help.

Also, I would like to know how the Mindsdb integration with DynamoDB works. When I query a model created on Mindsdb cloud, does Mindsdb reach out to Dynamodb to get the most up-to-date data in the database each time or continue using the data it retrieves from Dynamodb at the moment I linked the database to it? I really appreciate any help you can provide.

**Comment by bcs-gbl** [comment]
Hi Ghost,

1. First, you need to deploy MindsDB into the cloud, locally, etc

2. Set DynamoDB as your "source database":
```
CREATE DATABASE dynamodb_datasource       --- display name for the database
WITH ENGINE='dynamodb',                   --- name of the MindsDB handler
PARAMETERS={
  "aws_access_key_id": " ",               --- the AWS access key
  "aws_secret_access_key": " ",           --- the AWS secret access key
  "region_name": " "                      --- the AWS region
};
```
3. AWS side: Always generate different keys (and access roles) for different accesses, usage and services.

MindsDB under the hood use python+boto3. Maybe you should start from this info.

(For example: if you want to access dynamodb thru appsync (graphql), you can create a lambda resolver with boto3, and access to dynamodb. No need to modify @auth directive settings in your .graphql file. You can create lambda resolvers inside "Amplify": with "add function" command / advanced options))

**Docs:**
- https://docs.mindsdb.com/data-integrations/amazon-dynamodb
- https://docs.amplify.aws/cli/graphql/custom-business-logic/#lambda-function-resolver
- https://mindsdb.github.io/mindsdb/sql/create/databases/?h=dynamodb#dynamodb
